conans/test/integration/configuration/test_profile_jinja.py

```diff
@@ -105,3 +106,40 @@ def test_profile_version():
     client.run("install . -pr=profile1.jinja")
     assert f"*:myoption={conan_version}" in client.out
     assert "*:myoption2=True" in client.out
+
+
+def test_profile_template_profile_name():
+    """
+        The property profile_name should be parsed as the profile file name when rendering profiles
+    """
+    client = TestClient()
+    tpl1 = textwrap.dedent("""
+        [conf]
+        user.profile:name = {{ profile_name }}
+        """)
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        class Pkg(ConanFile):
+            def configure(self):
+                self.output.info("PROFILE NAME: {}".format(self.conf.get("user.profile:name")))
+        """)
+    client.save({"conanfile.py": conanfile,
+                 "profile_folder/foobar": tpl1,
+                 "another_folder/foo.profile": tpl1,
+                 os.path.join(client.cache_folder, "profiles", "baz"): tpl1})
+
+    # show only file name as profile name
+    client.run("install . -pr=profile_folder/foobar")
+    assert "conanfile.py: PROFILE NAME: foobar" in client.out
+
+    # profile_name should include file extension
+    client.run("install . -pr=another_folder/foo.profile")
+    assert "conanfile.py: PROFILE NAME: foo.profile" in client.out
+
+    # default profile should not compute profile_name by default
+    client.run("install . -pr=default")
+    assert "conanfile.py: PROFILE NAME: None" in client.out
```

Comment:
why not? I would expect the profile name to be ``PROFILE NAME: default``

Reply:
sorry, I was not clear on this test. We are using a custom conf to store the `profile_name`, but the profile `default` is totally clear, so we store nothing.

I just updated it now to provide the default profile with the same content + os=Windows (because the 'build' profile must have a 'os' declared)